src/generic/datavgen.cpp

```diff
@@ -2176,12 +2226,18 @@ wxDragResult wxDataViewMainWindow::OnData( wxDataFormat format, wxCoord x, wxCoo
     if ( row < GetRowCount() && xx <= GetEndOfLastCol() )
         item = GetItemByRow( row );
 
-    wxCustomDataObject *obj = (wxCustomDataObject *) GetDropTarget()->GetDataObject();
+    wxDataObjectComposite *obj = static_cast<wxDataObjectComposite*>(GetDropTarget()->GetDataObject());
+
+    size_t dataSize = obj->GetDataSize(format);
+    wxMemoryBuffer buffer;
+    obj->GetDataHere(format, buffer.GetWriteBuf(dataSize));
+    buffer.UngetWriteBuf(dataSize);
 
     wxDataViewEvent event(wxEVT_DATAVIEW_ITEM_DROP, m_owner, item);
     event.SetDataFormat( format );
-    event.SetDataSize( obj->GetSize() );
-    event.SetDataBuffer( obj->GetData() );
+    event.SetDataSize(dataSize);
+    event.SetDataBuffer(buffer.GetData());
+    event.SetDataObject(obj);
```

Comment:
I wonder if it could be better to pass `obj->GetObject(format)` here.

Reply:
In wxOSX port `wxCompositeObject` is passed into event, so I decided made it the same way in Generic for compatibility, but agree, it is better to pass `obj->GetObject(format)`.